Caboodle/FileSystem/FileSystem.ios.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Foundation;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class FileSystem
+    {
+        public static string CacheDirectory
+            => GetDirectory(NSSearchPathDirectory.CachesDirectory);
+
+        public static string AppDataDirectory
+            => GetDirectory(NSSearchPathDirectory.LibraryDirectory);
+
+        public static string UserDataDirectory
+            => GetDirectory(NSSearchPathDirectory.DocumentDirectory);
+
+        public static Task<Stream> OpenAppPackageFileAsync(string filename)
+        {
+            if (filename == null)
+                throw new ArgumentNullException(nameof(filename));
+
+            filename = filename.Replace("\\", "/");
+            var file = Path.Combine(NSBundle.MainBundle.BundlePath, filename);
```

Comment:
I prefer to use `NSBundle.ResourcePath` and `PathForResource` since that is the more consistent API between iOS and Mac. Bundles on iOS just happened to put their resources in the `BundlePath` but they are not there in Mac so the above code would fail.

https://developer.xamarin.com/api/property/MonoTouch.Foundation.NSBundle.ResourcePath/

https://developer.xamarin.com/api/member/MonoTouch.Foundation.NSBundle.PathForResource/p/System.String/System.String/

But since you are abstracting the API, it doesn't really matter - you can just write different versions for iOS and Mac.